code/game/objects/items/weapons/storage/holster.dm

```diff
@@ -0,0 +1,266 @@
+		/obj/item/gun/projectile/revolver,
+		/obj/item/gun/projectile/automatic/molly,
+		/obj/item/gun/projectile/paco,
+		/obj/item/gun/projectile/shotgun/doublebarrel/sawn, //short enough to fit in
+		/obj/item/gun/launcher/syringe,
+		/obj/item/gun/energy/plasma/brigador,
+		/obj/item/gun/projectile/shotgun/pump/sawn,
+		/obj/item/gun/projectile/boltgun/obrez,
+		/obj/item/gun/energy/retro/sawn,
+		/obj/item/gun/projectile/automatic/luty,
+		// /obj/item/gun/projectile/revolver/hornet
+		)
+
+	sliding_behavior = TRUE
+
+/obj/item/storage/pouch/holster/baton
+	name = "baton sheath"
+	desc = "Can hold a baton, or indeed most weapon shafts."
+	icon_state = "baton_holster"
+	item_state = "baton_holster"
+	rarity_value = 50
+
+	storage_slots = 1
+	max_w_class = ITEM_SIZE_BULKY
+
+	can_hold = list(
+		/obj/item/melee,
+		/obj/item/tool/crowbar,
+		/obj/item/tool/hammer/,
+		/obj/item/tool/hatchet
+		)
+
+	sliding_behavior = TRUE
+
+/obj/item/storage/pouch/holster/belt
+	name = "belt holster"
+	desc = "Can hold two handguns in. Quick on the draw!"
+	icon_state = "belt_holster"
+	item_state = "belt_holster"
+	rarity_value = 40
+
+	slot_flags = SLOT_BELT | SLOT_DENYPOCKET
+	price_tag = 240
+	matter = list(MATERIAL_BIOMATTER = 5)
+
+	sound_in = 'sound/effects/holsterin.ogg'
+	sound_out = 'sound/effects/holsterout.ogg'
+
+	max_w_class = ITEM_SIZE_HUGE
+
+	storage_slots = 2
+
+//Sheath
+/obj/item/storage/pouch/holster/belt/sheath
+	name = "sheath"
+	desc = "A sturdy brown leather sheath with a gold trim, made to house a variety of swords."
+	icon = 'icons/obj/sheath.dmi'
+	icon_state = "sheath_0"
+	item_state = "sheath_0"
+
+	price_tag = 120
+
+	storage_slots = 1
+
+	sound_in = 'sound/effects/sheathin.ogg'
+	sound_out = 'sound/effects/sheathout.ogg'
+
+	can_hold = list(
+		/obj/item/tool/sword/nt,
+		/obj/item/tool/sword/nt_sword,
+		/obj/item/tool/sword/saber,
+		/obj/item/tool/sword/katana,
+		/obj/item/tool/sword/katana/nano,
+		/obj/item/tool/sword,
+		/obj/item/toy/katana,
+		/obj/item/tool/sword/improvised
+		)
+	cant_hold = list(
+		/obj/item/tool/knife/dagger/nt,
+		/obj/item/tool/sword/nt/halberd,
+		/obj/item/tool/sword/nt/spear
+		)
+
+/obj/item/storage/pouch/holster/belt/sheath/improvised
+	name = "makeshift sheath"
+	desc = "A sturdy metal sheath with a rough finish. There's killing to do, draw your junkblade."
+	icon_state = "sheath_scrapsword"
+	price_tag = 50
+	matter = list(MATERIAL_STEEL = 2, MATERIAL_PLASTIC = 1)
+	can_hold = list(/obj/item/tool/sword/improvised)
+
+//Accessory holsters
+/obj/item/clothing/accessory/holster
+	name = "concealed carry holster"
+	desc = "An inconspicious holster that can be attached to your uniform. Can only fit small handguns and knives... Maybe something else, too."
+	icon_state = "concealed_carry"
+	slot = "utility"
+	matter = list(MATERIAL_BIOMATTER = 5)
+	price_tag = 160
+	spawn_blacklisted = FALSE
+	spawn_tags = SPAWN_TAG_HOLSTER
+	var/obj/item/storage/internal/holster
+	var/list/can_hold = list(
```

Comment:
Not being able to `H` your pistol out of this holster is already enough of a downside. Make it able to store same guns as pouch one. Don't forget to edit the description.

Reply:
Players expressed anger at not having what we(with maik) say is a free storage slot. Storage space is huge in "normal" Ss13 and so in eris. Maik even said that, I quote, "3 or 4 slots as a webbing means 4 more magazines, which are huge. An entire magazine pouch can only hold 6, and a tool pouch holds 4 tools too. Why a webbing that's free storage that lets you storage creep more?"

Back to the holster, however. Valo said that "a dude with a holdout" is fine. Even the others said that the attachment holster was used to have a _backup_ gun, to pull out in emergency. Dropped the gun, ran out of ammo, arrested. We give them what they want: the tactical belt already can hold pistols, but only "holdout" pistols: anne, makarov, giskard, and martin. So we keep the armpit holster as a concealed carry option. You can fit the small guns in it, which are good, but not powerful and can still kill plenty. 
But the other pistols and obrezes and sawn offs should not fit in the attachment. They're too good to be carried in a free storage. You have to make sacrifices for a 20 damage rifle. Why not a miller that does 50? You know that people even gloat about carrying a hidden Double-Barrel and instant point blank kill people with it because you can hide it for free in the suit slot. Now you need to sacrifice a single pouch slot for this, which might be an unoccupied suit slot anyway, or carry a holdout gun with their drawbacks.

Reply:
We dont want a shitty dollar store replacment for a uniform holster that worked, we want the orginal thing.

You keep pretending  not to understand even though public opinion is about liking uniform holsters.

A shitty sword or a dinky sword being presented as a consolation prize is actively insulting